include/mesh.h

```diff
@@ -447,6 +448,33 @@ class Mesh {
   // Initialise the nodal properties' map
   void initialise_nodal_properties();
 
+  //! Set particles lsm values
+  void assign_particle_levelset(std::vector<double>& phi_list);
+
+  // Create the nodal properties' map for discontinuity
+  void create_nodal_properties_discontinuity();
+
+  //! Initialise discontinuities
+  //! \param[in] discontinuities
+  void initialise_discontinuities(
+      const std::map<unsigned, std::shared_ptr<mpm::DiscontinuityBase<Tdim>>>&
+          discontinuities) {
+    discontinuities_ = discontinuities;
+  }
+
+  //! Locate points of discontinuity in a cell
+  void locate_discontinuity_mesh();
```

Comment:
```suggestion
  void locate_discontinuity();
```
Mesh would be redundant here